space-cli/cmd/modules/letsencrypt/generate_test.go

```diff
@@ -0,0 +1,105 @@
+package letsencrypt
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spaceuptech/space-cli/cmd/model"
+	"github.com/spaceuptech/space-cli/cmd/utils"
+	"github.com/spaceuptech/space-cli/cmd/utils/input"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_generateLetsEncryptDomain(t *testing.T) {
+	someString := ""
+	type mockArgs struct {
+		method         string
+		args           []interface{}
+		paramsReturned []interface{}
+	}
+	tests := []struct {
+		name           string
+		surveyMockArgs []mockArgs
+		want           *model.SpecObject
+		wantErr        bool
+	}{
+		{
+			name: "error surveying white listed domains",
+			surveyMockArgs: []mockArgs{
+				{
+					method:         "AskOne",
+					args:           []interface{}{&survey.Input{Message: "Enter White Listed Domain by comma seperated value: "}, &someString, mock.Anything},
+					paramsReturned: []interface{}{errors.New("some error"), ""},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "error surveying project",
+			surveyMockArgs: []mockArgs{
+				{
+					method:         "AskOne",
+					args:           []interface{}{&survey.Input{Message: "Enter White Listed Domain by comma seperated value: "}, &someString, mock.Anything},
+					paramsReturned: []interface{}{nil, ""},
+				},
+				{
+					method:         "AskOne",
+					args:           []interface{}{mock.Anything, &someString, mock.Anything},
+					paramsReturned: []interface{}{errors.New("some error"), ""},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "no error surveying anything",
+			surveyMockArgs: []mockArgs{
+				{
+					method:         "AskOne",
+					args:           []interface{}{&survey.Input{Message: "Enter White Listed Domain by comma seperated value: "}, &someString, mock.Anything},
+					paramsReturned: []interface{}{nil, "domain1,domain2"},
+				},
+				{
+					method:         "AskOne",
+					args:           []interface{}{mock.Anything, &someString, mock.Anything},
+					paramsReturned: []interface{}{nil, ""},
+				},
+			},
+			want: &model.SpecObject{
+				API:  "/v1/config/projects/{project}/letsencrypt/config/{id}",
+				Type: "letsencrypt",
+				Meta: map[string]string{
+					"project": "",
+					"id":      "letsencrypt-config",
+				},
+				Spec: map[string]interface{}{
+					"domains": []string{"domain1", "domain2"},
+				},
+			},
```

Comment:
And one more functionality in generateletsencrypt. Ask for email id after getting the domain name. Add the above in test also.
Check letsencrypt config struct in gateway for more info